### DISCUSSION TITLE
Fixing Malformed Archives Before Exports

### DISCUSSION THREAD

**Discussion by lifely** [fastlane/fastlane#19777] [Show and tell]
Hello,

I've been working on a projects w/ multiple targets and dependencies for applications and extensions.
By choice, Cocoapods been used for the project and thus I've encountered issue w/ multiple-targets and dependencies.

This issue appears on the new build system while using cocoapods subspecs as described here:
> https://github.com/CocoaPods/CocoaPods/issues/8206

If you follow the thread you'll see that I've found a few solutions for this problem that resolve all issue and product a `.xcarchive` for release build.

However to achieve this release we also need to include some `framework` in the archive content.
This create an `incomplete` archive `plist` which doesn't include the `ApplicationProperties` key.

However this `ApplicationProperties` is required for xcode or command line tools to export the final app or ipa for upload to apple or any other need.

The archive is actually completely valid, it's simply contains multiple products, identifying the correct binary and adding the `ApplicationProperties` to the archive plist fix this.

So I've decided to solve this by scripting, and Fastlane `Gym::Runner: verify_archive` was the best place for this.
Currently I've implemented this as a quick fix in my own `FastFile` but maybe this could be baked into Fastlane as is.

What do you guys think, you can find my solution here:
https://gist.github.com/lifely/7f8b68bb3f1fed31da8c1a8aff93ae6b

If needed I can provided a pull request, not sure if it's great, any recommendation ?

**Comment by lucgrabowski** [comment]
@joshdholtz Could you take a look at this idea ;)?
